File: src/main/java/com/github/morningzeng/toolset/ui/enums/CodingEnum.java
package com.github.morningzeng.toolset.ui.enums;

import com.github.morningzeng.toolset.ui.Base64Component;
import com.github.morningzeng.toolset.ui.URLComponent;
import com.intellij.openapi.project.Project;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.experimental.Accessors;

import javax.swing.Icon;
import javax.swing.JComponent;

/**
 * @author Morning Zeng
 * @since 2024-05-21
 */
@Getter
@AllArgsConstructor
@Accessors(fluent = true)
public enum CodingEnum implements TabSupport {

    BASE64("Base64", null, "Base64 Encode and Decode") {
        @Override
        public JComponent component(final Project project) {
            return new Base64Component(project);
        }
    },
    URL("URL", null, "URL Encode and Decode") {
        @Override
        public JComponent component(final Project project) {
            return new URLComponent(project);
        }
    },
    ;


    private final String title;
    private final Icon icon;
    private final String tips;

}

File: src/main/kotlin/com/github/morningzeng/toolbox/ui/view/coding/CodingEnum.kt
package com.github.morningzeng.toolbox.ui.view.coding

import com.github.morningzeng.toolbox.ui.TabSupport
import com.intellij.openapi.wm.ToolWindow
import javax.swing.Icon
import javax.swing.JComponent

/**
 * @author Morning Zeng
 * @since 2025-07-22
 */
enum class CodingEnum(
    override val title: String,
    override val icon: Icon? = null,
    override val tips: String
) : TabSupport {

    BASE64("Base64", null, "Base64 Encode & Decode") {
        override fun component(toolWindow: ToolWindow): JComponent = Base64Component(toolWindow.project)
    },
    URL("URL", null, "URL Encode & Decode") {
        override fun component(toolWindow: ToolWindow): JComponent = URLComponent(toolWindow.project)
    },

    ;

}